NitroxClient/GameLogic/Entities.cs

```diff
@@ -166,6 +171,24 @@ private void AddPendingParentEntity(Entity entity)
             pendingEntities.Add(entity);
         }
 
+        // Nitrox uses entity spawners to generate the various gameObjects in the world. These spawners are invoked using 
+        // IEnumerator (async) and levarage async Prefab/CraftData instantiation functions.  However, even though these
+        // functions are successful, it doesn't mean the entity is fully setup.  Subnautica is known to spawn coroutines 
+        // in the start() method of objects to spawn prefabs or other objects. An example is anything with a battery, 
+        // which gets configured after the fact.  In most cases, Nitrox needs to wait for objets to be fully spawned in 
+        // order to setup ids.  Historically we would persist metadata and use a patch to later tag the item, which gets
+        // messy.  This function will allow us wait on any type of instantiation necessary; this can be optimized later
+        // to move on to other spawning and come back when this item is ready.  
+        private IEnumerator AwaitAnyRequiredEntitySetup(GameObject gameObject)
+        {
+            EnergyMixin energyMixin = gameObject.GetComponent<EnergyMixin>();
+
+            if (energyMixin)
+            {
+                yield return new WaitUntil(() => energyMixin.battery != null);
```

Comment:
I guess energyMixin.battery is a MonoBehaviour?
```suggestion
                yield return new WaitUntil(() => energyMixin.battery);
```

Reply:
sounds good.

Reply:
actually this is not a mono :)